lang/CMakeLists.txt

```diff
@@ -252,7 +252,25 @@ endif()
 
 if (SC_ABLETON_LINK)
 	message(STATUS "Compiling with Ableton Link support")
-	include(../external_libraries/link/AbletonLinkConfig.cmake)
+	if(SYSTEM_ABLETON_LINK)
+		find_path(ABLETON_LINK_INCLUDE ableton Link.hpp)
+		add_library(Ableton::Link IMPORTED INTERFACE)
+		set_property(TARGET Ableton::Link APPEND PROPERTY
+			INTERFACE_INCLUDE_DIRECTORIES
+			${ABLETON_LINK_INCLUDE}/ableton
+		)
+		set_property(TARGET Ableton::Link APPEND PROPERTY
+			INTERFACE_COMPILE_DEFINITIONS
+			LINK_PLATFORM_LINUX=1
+		)
+		set_property(TARGET Ableton::Link APPEND PROPERTY
+			INTERFACE_SOURCES
+			${ABLETON_LINK_INCLUDE}/ableton/Link.hpp
+		)
+	else()
+		include(../external_libraries/link/AbletonLinkConfig.cmake)
```

Comment:
shouldn't this be included in a package distribution of Link? IIUC that is the correct way to distribute and use libraries with CMake support

Reply:
Hm, maybe you're right. [This is what Debian ships](https://packages.debian.org/buster/all/ableton-link-dev/filelist).

I have  not included the cmake files in packaging, because I didn't know where to put them.
Would `/usr/share/ableton-link/` or `/usr/share/link` by the correct place for the .cmake files? Is there a standard for this?

Reply:
That being said: [Debian also patches some of the files](https://salsa.debian.org/multimedia-team/ableton-link/-/blob/master/debian/patches/DEBIAN_system_libraries.patch) to make it compatible with a system asio (I do a similar thing, but ignore the .cmake files later on).

Reply:
> Is there a standard for this?

AFAIK there is no single best practice, but a couple things that will work. according to the documentation for [find_package](https://cmake.org/cmake/help/v3.0/command/find_package.html), the following paths are tried with Linux:

```
<prefix>/(lib/<arch>|lib|share)/cmake/<name>*/          (U)
<prefix>/(lib/<arch>|lib|share)/<name>*/                (U)
<prefix>/(lib/<arch>|lib|share)/<name>*/(cmake|CMake)/  (U)
```

where `<name>` is Link if `find_package()` is called with "Link".

if you call find_package without NO_SYSTEM_ENVIRONMENT_PATH then `<prefix>` will eventually be tried with `/usr/`. so a lot of potential options here. let me see if my copy of Professional CMake has a recommended best practice ;)

Reply:
btw, on my Arch system I see this:

<details>

```
[bheim@archlinux supercollider]$ find /usr -name *.cmake | grep -v /usr/lib/cmake | grep -v /usr/share/cmake | sort | less

/usr/lib32/cmake/DBus1/DBus1Config.cmake
/usr/lib32/cmake/DBus1/DBus1ConfigVersion.cmake
/usr/lib32/cmake/harfbuzz/harfbuzz-config.cmake
/usr/lib32/cmake/libxml2/libxml2-config.cmake
/usr/lib32/cmake/Ogg/OggConfig.cmake
/usr/lib32/cmake/Ogg/OggConfigVersion.cmake
/usr/lib32/cmake/Ogg/OggTargets.cmake
/usr/lib32/cmake/Ogg/OggTargets-noconfig.cmake
/usr/lib32/cmake/PulseAudio/PulseAudioConfig.cmake
/usr/lib32/cmake/PulseAudio/PulseAudioConfigVersion.cmake
/usr/lib/CLucene/CLuceneConfig.cmake
/usr/lib/openjpeg-2.3/OpenJPEGConfig.cmake
/usr/lib/openjpeg-2.3/OpenJPEGTargets.cmake
/usr/lib/openjpeg-2.3/OpenJPEGTargets-release.cmake
/usr/local/clang-8/lib/cmake/clang/ClangConfig.cmake
/usr/local/clang-8/lib/cmake/clang/ClangTargets.cmake
/usr/local/clang-8/lib/cmake/clang/ClangTargets-release.cmake
/usr/local/clang-8/lib/cmake/llvm/AddLLVM.cmake
/usr/local/clang-8/lib/cmake/llvm/AddLLVMDefinitions.cmake
/usr/local/clang-8/lib/cmake/llvm/AddOCaml.cmake
/usr/local/clang-8/lib/cmake/llvm/AddSphinxTarget.cmake
/usr/local/clang-8/lib/cmake/llvm/CheckAtomic.cmake
/usr/local/clang-8/lib/cmake/llvm/CheckCompilerVersion.cmake
/usr/local/clang-8/lib/cmake/llvm/CheckLinkerFlag.cmake
/usr/local/clang-8/lib/cmake/llvm/ChooseMSVCCRT.cmake
/usr/local/clang-8/lib/cmake/llvm/CrossCompile.cmake
/usr/local/clang-8/lib/cmake/llvm/DetermineGCCCompatible.cmake
/usr/local/clang-8/lib/cmake/llvm/FindLibpfm.cmake
/usr/local/clang-8/lib/cmake/llvm/FindOCaml.cmake
/usr/local/clang-8/lib/cmake/llvm/FindSphinx.cmake
/usr/local/clang-8/lib/cmake/llvm/GenerateVersionFromCVS.cmake
/usr/local/clang-8/lib/cmake/llvm/GetSVN.cmake
/usr/local/clang-8/lib/cmake/llvm/HandleLLVMOptions.cmake
/usr/local/clang-8/lib/cmake/llvm/HandleLLVMStdlib.cmake
/usr/local/clang-8/lib/cmake/llvm/LLVM-Config.cmake
/usr/local/clang-8/lib/cmake/llvm/LLVMConfig.cmake
/usr/local/clang-8/lib/cmake/llvm/LLVMConfigVersion.cmake
/usr/local/clang-8/lib/cmake/llvm/LLVMExports.cmake
/usr/local/clang-8/lib/cmake/llvm/LLVMExports-release.cmake
/usr/local/clang-8/lib/cmake/llvm/LLVMExternalProjectUtils.cmake
/usr/local/clang-8/lib/cmake/llvm/LLVMInstallSymlink.cmake
/usr/local/clang-8/lib/cmake/llvm/LLVMProcessSources.cmake
/usr/local/clang-8/lib/cmake/llvm/TableGen.cmake
/usr/local/clang-8/lib/cmake/llvm/VersionFromVCS.cmake
/usr/share/doc/gc/README.cmake
/usr/share/graphite2/graphite2.cmake
/usr/share/graphite2/graphite2-release.cmake
```

</details>